pkg/pool-aclamm/contracts/lib/AclAmmMath.sol

```diff
@@ -145,6 +173,23 @@ library AclAmmMath {
         }
     }
 
+    function calculateSqrtQ0(
+        uint256 currentTime,
+        uint256 startSqrtQ0,
+        uint256 endSqrtQ0,
+        uint256 startTime,
+        uint256 endTime
+    ) internal pure returns (uint256) {
+        if (currentTime >= endTime) {
```

Comment:
In this function, we should probably check if currentTime < startTime and return startQ0. This condition should be tested.